File: BackEnd.Net/src/Alcohol/Models/Recipe.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Alcohol.Models
{
    public class Recipe
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [StringLength(500)]
        public string Description { get; set; }

        [Required]
        [StringLength(100)]
        public string Slug { get; set; }

        [StringLength(500)]
        public string ImageUrl { get; set; }

        [Required]
        public string Instructions { get; set; }

        [Required]
        [StringLength(50)]
        public string Difficulty { get; set; }

        [Required]
        public int PreparationTime { get; set; }

        [Required]
        public int Servings { get; set; }

        [Required]
        public int DisplayOrder { get; set; }

        [Required]
        public bool IsActive { get; set; }

        [StringLength(200)]
        public string MetaTitle { get; set; }

        [StringLength(500)]
        public string MetaDescription { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Changed from many-to-many to direct relationship
        public int? CategoryId { get; set; }
        public Category Category { get; set; }

        // Navigation properties
        public ICollection<RecipeIngredient> Ingredients { get; set; }

        public Recipe()
        {
            CreatedAt = DateTime.UtcNow;
            Ingredients = new HashSet<RecipeIngredient>();
        }
    }

    public class RecipeIngredient
    {
        [Key]
        public int Id { get; set; }

        public int RecipeId { get; set; }
        [ForeignKey("RecipeId")]
        public Recipe Recipe { get; set; }

        public int? ProductId { get; set; }
        [ForeignKey("ProductId")]
        public Product Product { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        public decimal Quantity { get; set; }

        [StringLength(50)]
        public string Unit { get; set; }

        public string Notes { get; set; }
    }

    public class RecipeCategory
    {
        [Key]
        public int Id { get; set; }

        public int RecipeId { get; set; }
        [ForeignKey("RecipeId")]
        public Recipe Recipe { get; set; }

        public int CategoryId { get; set; }
        [ForeignKey("CategoryId")]
        public Category Category { get; set; }
    }
} 

File: FrontEndNext/frontendnext/src/lib/types/recipe.ts
export interface Recipe {
  id: number;
  name: string;
  description: string;
  instructions: string;
  difficulty: string;
  prepTime: number;
  cookTime: number;
  servings: number;
  imageUrl: string;
  isFeatured: boolean;
  createdAt: string;
  updatedAt?: string;
  categories?: RecipeCategory[];
  ingredients?: RecipeIngredient[];
}

export interface RecipeCategory {
  id: number;
  name: string;
  description: string;
}

export interface RecipeIngredient {
  id: number;
  recipeId: number;
  productId?: number;
  name: string;
  quantity: number;
  unit?: string;
  notes?: string;
  product?: { id: number; name: string; imageUrl: string }; // Product type for ingredient
}

export interface RecipeCreateDto {
  name: string;
  description: string;
  instructions: string;
  prepTime: number;
  cookTime: number;
  servings: number;
  difficulty: string;
  imageUrl: string;
  isFeatured: boolean;
  categoryIds: number[];
  ingredients: RecipeIngredientCreateDto[];
}

export interface RecipeUpdateDto {
  name: string;
  description: string;
  instructions: string;
  prepTime: number;
  cookTime: number;
  servings: number;
  difficulty: string;
  imageUrl: string;
  isFeatured: boolean;
  categoryIds: number[];
  ingredients: RecipeIngredientUpdateDto[];
}

export interface RecipeIngredientCreateDto {
  name: string;
  quantity: number;
  unit: string;
  notes?: string;
  productId?: number;
}

export interface RecipeIngredientUpdateDto {
  id: number;
  name: string;
  quantity: number;
  unit: string;
  notes?: string;
  productId?: number;
}
